src/Oscoin/Crypto/PubKey.hs

```diff
@@ -49,33 +51,37 @@ class HasDigitalSignature c where
     data family Signature c :: *
 
     sign   :: (ByteArrayAccess msg, MonadRandom m) => PrivateKey c -> msg -> m (Signed c msg)
-    verify :: ByteArrayAccess msg => PublicKey c -> Signed c msg -> Bool
+    verify :: ByteArrayAccess msg => Signed c msg -> Bool
 
     -- | Generate a new random keypair.
     generateKeyPair :: (Crypto.HasHashing c, MonadRandom m) => m (KeyPair c)
 
+    -- | Derive the public key of a private key.
+    derivePublicKey :: PrivateKey c -> PublicKey c
```

Comment:
This seems generally useful, however we don't need `generateKeyPair` as a class method anymore. It should be replaced by `generateSecretKey`, and a top-level convenience function `generateKeyPair` could be provided which is implemented in terms of `generateSecretKey` and `derivePublicKey`.

Reply:
Good point!

Reply:
Hmm annoyingly, cryptonite doesn't expose a `generatePrivateKey`, meaning our implementation of it would have to generate a pair and discard the public key (which maybe will never be computed due to laziness :thinking: )